src/main/java/org/folio/rs/service/RetrievalQueueService.java

```diff
@@ -96,4 +138,52 @@ private void saveRetrievalQueueWithCurrentDate(RetrievalQueueRecord record) {
     return (record, criteria, builder) -> builder.equal(record.get(ID), stringToUUIDSafe(id));
   }
 
+  private LocationMapping getLocationMapping(Item item) {
+    return Objects.nonNull(item.getEffectiveLocation())
+        ? locationMappingsService.getMappingByFolioLocationId(item.getEffectiveLocation().getId())
+        : null;
+  }
+
+  private Item getOriginalItemByBarcode(MovedEventRequest movedEventRequest) {
+    ResultList<Item> items = inventoryClient.getItem("barcode==" + movedEventRequest.getItemBarCode());
+    if (Objects.isNull(items)) {
```

Comment:
Double check please this functionality in case if users will be empty but dont null